src/io/input_split_base.cc

```diff
@@ -226,12 +230,18 @@ bool InputSplitBase::ReadChunk(void *buf, size_t *size) {
                             max_size - olen);
   nread += olen;
   if (nread == 0) return false;
-  if (nread == olen) {  // add extra newline to handle files with NOEOL
-    char *bufptr = reinterpret_cast<char*>(buf);
-    bufptr[nread] = '\n';
-    nread++;
+  if (this->IsTextParser()) {
+    if (nread == olen) {  // add extra newline to handle files with NOEOL
+      char *bufptr = reinterpret_cast<char*>(buf);
+      bufptr[nread] = '\n';
```

Comment:
comment explaining why \b here again. In what case this \n will be helpful compared to previous \n

Reply:
I'll add a link to the issue that gave rise to it.